src/components/scenes/TradeCreateScene.tsx

```diff
@@ -0,0 +1,762 @@
+                <StepText numberOfLines={0}>
+                  {lstrings.trade_region_select_step_3}
+                </StepText>
+              </StepRow>
+              <StepRow>
+                <StepNumberText>
+                  {sprintf(lstrings.step_prefix_s, '4')}
+                </StepNumberText>
+                <StepText numberOfLines={0}>
+                  {lstrings.trade_region_select_step_4}
+                </StepText>
+              </StepRow>
+            </StepContainer>
+          </StepsCard>
+
+          <RegionButton onPress={handleRegionSelect}>
+            <RegionButtonContent>
+              {flagUri != null ? (
+                <FlagIcon source={{ uri: flagUri }} sizeRem={1.5} />
+              ) : (
+                <GlobeIcon
+                  name="globe"
+                  color={theme.iconTappable}
+                  size={theme.rem(1.5)}
+                />
+              )}
+              <RegionButtonText>{getRegionText()}</RegionButtonText>
+              <Feather
+                name="chevron-right"
+                color={theme.iconTappable}
+                size={theme.rem(1.25)}
+              />
+            </RegionButtonContent>
+          </RegionButton>
+        </SceneContainer>
+      </SceneWrapper>
+    )
+  }
+
+  // Render trade form view
+  return (
+    <SceneWrapper scroll>
+      <SceneContainer
+        headerTitle={lstrings.buy_crypto}
+        headerTitleChildren={
+          <PillButton
+            icon={() =>
+              flagUri != null ? <FlagIcon source={{ uri: flagUri }} /> : null
+            }
+            label={getRegionText()}
+            onPress={handleRegionSelect}
+          />
+        }
+      >
+        {/* Amount Inputs */}
+        <InputsContainer>
+          {/* Top Input (Fiat by design) */}
+          <InputRow>
+            <DropDownInputButton onPress={handleFiatDropdown}>
+              {selectedFiatFlagUri !== '' ? (
+                <FlagIcon sizeRem={1.5} source={{ uri: selectedFiatFlagUri }} />
+              ) : (
+                <FiatIcon
+                  sizeRem={1.5}
+                  fiatCurrencyCode={selectedFiatCurrencyCode}
+                />
+              )}
+            </DropDownInputButton>
+
+            <InputContainer>
+              <FilledTextInput
+                value={displayFiatAmount}
+                onChangeText={handleFiatChangeText}
+                placeholder={sprintf(
+                  lstrings.trade_create_amount_s,
+                  selectedFiatCurrencyCode
+                )}
+                keyboardType="decimal-pad"
+                numeric
+                showSpinner={isFetchingQuotes && lastUsedInput === 'crypto'}
+              />
+            </InputContainer>
+          </InputRow>
+
+          {/* Bottom Input (Crypto by design) */}
+          <InputRow>
+            <DropDownInputButton onPress={handleCryptDropdown}>
+              {selectedCrypto == null || selectedWallet == null ? (
```

Comment:
When no wallet is selected, rendering FlagIcon with an empty URI may show a broken image; consider a placeholder or null.